src/main/java/com/minecolonies/coremod/util/IItemHandlerToIInventoryWrapper.java

```diff
@@ -0,0 +1,172 @@
+package com.minecolonies.coremod.util;
+
+import com.minecolonies.coremod.inventory.api.IWorldNameableModifyable;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class wrapper for an IItemHandler to an IInventory.
+ */
+public class IItemHandlerToIInventoryWrapper implements IInventory
+{
+    private final IItemHandlerModifiable wrapped;
+    private final IWorldNameableModifyable named;
+
+    public IItemHandlerToIInventoryWrapper(final IItemHandlerModifiable wrapped, final IWorldNameableModifyable named) {this.wrapped = wrapped;
+        this.named = named;
+    }
+
+    /**
+     * Returns the number of slots in the inventory.
+     */
+    @Override
+    public int getSizeInventory()
+    {
+        return wrapped.getSlots();
+    }
+
+    /**
+     * Returns the stack in the given slot.
+     */
+    @Nullable
+    @Override
+    public ItemStack getStackInSlot(final int index)
+    {
+        return wrapped.getStackInSlot(index);
+    }
+
+    /**
+     * Removes up to a specified number of items from an inventory slot and returns them in a new stack.
+     */
+    @Nullable
+    @Override
+    public ItemStack decrStackSize(final int index, final int count)
+    {
+        return wrapped.extractItem(index, count, false);
+    }
+
+    /**
+     * Removes a stack from the given slot and returns it.
+     */
+    @Nullable
+    @Override
+    public ItemStack removeStackFromSlot(final int index)
+    {
+        return decrStackSize(index, Integer.MAX_VALUE);
+    }
+
+    /**
+     * Sets the given item stack to the specified slot in the inventory (can be crafting or armor sections).
+     */
+    @Override
+    public void setInventorySlotContents(final int index, @Nullable final ItemStack stack)
+    {
+        wrapped.setStackInSlot(index, stack);
+    }
+
+    /**
+     * Returns the maximum stack size for a inventory slot. Seems to always be 64, possibly will be extended.
+     */
+    @Override
+    public int getInventoryStackLimit()
+    {
+        return 64;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 64 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)